### ISSUE TITLE
Missing symlinked modules

### ISSUE THREAD

**Issue by heaths** [obi1kenobi/cargo-semver-checks#1718]
### Steps to reproduce the bug with the above code

1. Clone <https://github.com/Azure/azure-sdk-for-rust>
2. Run: `cargo semver-checks --manifest-path sdk/keyvault/azure_security_keyvault_secrets/Cargo.toml --baseline-rev azure_security_keyvault_secrets@1.0.0`

### Actual Behaviour

Fails to find module that is symlinked in the git repo:

```
error[E0583]: file not found for module `authorizer`
 --> /mnt/vss/_work/1/s/target/semver-checks/git-azure_security_keyvault_secrets_1_0_0/fd2cf10e8968554dafe01a113374bbd97b4eadbb/sdk/keyvault/azure_security_keyvault_secrets/src/lib.rs:7:1
  |
7 | mod authorizer;
  | ^^^^^^^^^^^^^^^
  |
```

This is symlinked from `sdk/keyvault/azure_security_keyvault_keys/src/authorizer.rs`.

### Expected Behaviour

Symlinked files should be checked out. Alternatively, couldn't crate files just be extracted? Why clone a repo at all when the crate contains all the files laid out as necessary with no additional cruft.

### Generated System Information

Note: this is not from the machine exhibiting the problem, but is the same version of rustc and cargo-semver-checks:

---

System information:
-------------------
#### Software version

cargo-semver-checks 0.50.0

#### Operating system

- OS: Linux (Ubuntu 24.04)
- Kernel: 6.18.33.2-microsoft-standard-WSL2

#### Command-line

```bash
/home/heaths/.cargo/bin/cargo-semver-checks semver-checks --bugreport
```

#### cargo version

```
> cargo -V
cargo 1.95.0 (f2d3ce0bd 2026-03-21)
```

#### Compile time information

- Profile: release
- Target triple: aarch64-unknown-linux-gnu
- Family: unix
- OS: linux
- Architecture: aarch64
- Pointer width: 64
- Endian: little
- CPU features: neon
- Host: aarch64-unknown-linux-gnu

Cargo build configuration:
--------------------------
[doc]
browser = ["wslview"]

Please file an issue on GitHub reporting your bug.
Consider adding the diagnostic information above, either manually or automatically through the link below:

https://github.com/obi1kenobi/cargo-semver-checks/issues/new?labels=C-bug&template=3-bug-report.yml&sys-info=%23%23%23%23%20Software%20version%0A%0Acargo-semver-checks%200.50.0%0A%0A%23%23%23%23%20Operating%20system%0A%0A-%20OS%3A%20Linux%20%28Ubuntu%2024.04%29%0A-%20Kernel%3A%206.18.33.2-microsoft-standard-WSL2%0A%0A%23%23%23%23%20Command-line%0A%0A%60%60%60bash%0A%2Fhome%2Fheaths%2F.cargo%2Fbin%2Fcargo-semver-checks%20semver-checks%20--bugreport%0A%60%60%60%0A%0A%23%23%23%23%20cargo%20version%0A%0A%60%60%60%0A%3E%20cargo%20-V%0Acargo%201.95.0%20%28f2d3ce0bd%202026-03-21%29%0A%60%60%60%0A%0A%23%23%23%23%20Compile%20time%20information%0A%0A-%20Profile%3A%20release%0A-%20Target%20triple%3A%20aarch64-unknown-linux-gnu%0A-%20Family%3A%20unix%0A-%20OS%3A%20linux%0A-%20Architecture%3A%20aarch64%0A-%20Pointer%20width%3A%2064%0A-%20Endian%3A%20little%0A-%20CPU%20features%3A%20neon%0A-%20Host%3A%20aarch64-unknown-linux-gnu&build-config=%5Bdoc%5D%0Abrowser%20%3D%20%5B%22wslview%22%5D%0A
